### PR TITLE
Spelling checker has been modified

### PR DESCRIPTION
**Pull request by bitanb1999**
Please check the options that you have completed and strike out the options that do not apply via this pull request:

- [X] a clear title and description of the Pull Request has been provided
 you have read
- [X]  the [Contributing doc](https://github.com/neomatrix369/nlp_profiler/blob/master/CONTRIBUTING.md)
- [X]  the [Developer Guide](https://github.com/neomatrix369/nlp_profiler/blob/master/developer-guide.md)
- [X]  the pull request passes the tests (`./test-coverage "tests slow-tests"``) - this will also be visible via the Code coverage report and CI/CD task on the Pull Request
- [X]  you have performed some kind of smoke test by running your changes in an isolated environment i.e. Docker container, Google Colab, Kaggle, etc...
- []  ~~the notebooks are updated (see notebooks folder, read the [Notebooks](https://github.com/neomatrix369/nlp_profiler/blob/master/.github/notebooks/README.md) docs)~~
- [X]  [CHANGELOG.md](https://github.com/neomatrix369/nlp_profiler/blob/master/CHANGELOG.md) has been updated (please follow the existing format)

### Goal or purpose of the PR

The spelling checker previously used TextBlob and required tokenization for the spelling checking and spelling quality summarisation. This took significant time and the result score calculated was also not satisfactory.

### Changes implemented in the PR

I replaced the checker function with a package that states to be much faster than TextBlob and jamspell, namely, Symspellpy. Further, the result scoring was entirely based on the ratio of the number of misspelled words to the total length of the string. This doesn't take ease of reading or "whether the phrase makes sense" into account. To resolve these issues, I used fuzzy-matching techniques that compare the original text and the rectified text and mark the score of the text accordingly.

### CODE REVIEW

File: nlp_profiler/high_level_features/ease_of_reading_check.py [hunk 1]
```diff
@@ -44,7 +44,11 @@ def ease_of_reading_score(text: str) -> float:
     if (not isinstance(text, str)) or (len(text.strip()) == 0):
         return NaN
 
-    return float(flesch_reading_ease(text))
+    score = float(flesch_reading_ease(text))
```
Comment (neomatrix369) [inline nlp_profiler/high_level_features/ease_of_reading_check.py:47]: good fix, please apply code formatting with the help of `black`

**Comment by neomatrix369** [comment]
This PR depends on the merging of PR #70 - once we merge that PR the current one can proceed but till then let's resolve any comments on this PR

**Comment by neomatrix369** [comment]
Please also do one last check in https://github.com/neomatrix369/nlp_profiler/blob/master/CONTRIBUTING.md to see if any dependent files need changing i.e. re-running notebooks etc, the [Developer Guide](https://github.com/neomatrix369/nlp_profiler/blob/master/developer-guide.md) is also something to review as a closing action.

Maybe you can enhance the existing grammar check example in the notebook(s) to illustrate the new package's features.

There are notebooks on this repo, please take a look at them and re-run them on your local machine to see if your changes have taken effect and no issues have arisen.

File: nlp_profiler/high_level_features/spelling_quality_check.py [hunk 2]
```diff
@@ -3,8 +3,25 @@
...
+sym_spell = SymSpell(max_dictionary_edit_distance=2, prefix_length=7)
+dictionary_path = pkg_resources.resource_filename(
+    "symspellpy", "frequency_dictionary_en_82_765.txt"
+)
+# term_index: column of the term 
+# count_index: column of the term's frequency
+sym_spell.load_dictionary(dictionary_path, term_index=0, count_index=1)
+
+#importing fuzzy matching for scoring purposes
+from fuzzywuzzy import fuzz
```
Comment (neomatrix369) [inline nlp_profiler/high_level_features/spelling_quality_check.py:23]: These two lines can be written as one line, comma-separated imports:
```python
from fuzzywuzzy import a, b
```

**Review by neomatrix369 — CHANGES_REQUESTED** [review]
LGTM - just a few changes requested

**Comment by neomatrix369** [comment]
This PR is related to https://github.com/neomatrix369/nlp_profiler/issues/8, have a good read of the issue to see if all or most of the requirements there are resolved by this PR

File: nlp_profiler/high_level_features/spelling_quality_check.py (hunk 2 above)
Comment (bitanb1999) [reply nlp_profiler/high_level_features/spelling_quality_check.py]: done!

**Comment by bitanb1999** [comment]
> This PR is related to #8, have a good read of the issue to see if all or most of the requirements there are resolved by this PR

I checked #8 and #2 and it addresses both issues. The results have been modified with fuzzy algorithm and they are penalizing for each misspelled word and arrangement of tokens. See this article: https://towardsdatascience.com/fuzzy-string-matching-in-python-68f240d910fe 
Also, the [Symspell](https://github.com/wolfgarbe/SymSpell) package is much faster than TextBlob as stated by multiple articles and hence #2  is also being addressed.

File: nlp_profiler/high_level_features/ease_of_reading_check.py (hunk 1 above)
Comment (neomatrix369) [reply nlp_profiler/high_level_features/ease_of_reading_check.py:47]: See this https://www.freecodecamp.org/news/auto-format-your-python-code-with-black/

File: nlp_profiler/high_level_features/grammar_quality_check.py [hunk 3]
```diff
@@ -26,11 +26,12 @@ def apply_grammar_check(heading: str,
 ### Grammar check: this is a very slow process
 ### take a lot of time per text it analysis
 def grammar_check_score(text: str) -> int:
-    if (not isinstance(text, str)) or (len(text.strip()) == 0):
+    if not isinstance(text, str) or not text.strip():
```
Comment (neomatrix369) [inline nlp_profiler/high_level_features/grammar_quality_check.py:29]: The two `not`s can be combined:
```python
if not (isinstance(text, str) or text.strip()):
```

**Comment by neomatrix369** [comment]
One last thing to do is update the [CHANGELOG.md](https://github.com/neomatrix369/nlp_profiler/blob/master/CHANGELOG.md) for this change - its very easy to do, see how the previous ones are done

**Pull request merged**